x-pack/plugin/src/test/resources/rest-api-spec/api/ml.stop_datafeed.json

```diff
@@ -22,6 +22,11 @@
       ]
     },
     "params":{
+      "allow_no_match":{
+        "type":"boolean",
+        "required":false,
+        "description":"Whether to ignore if a wildcard expression matches no datafeeds. (This includes `_all` string or when no datafeeds have been specified)"
+      },
       "allow_no_datafeeds":{
```

Comment:
Is there a "deprecation" field or something for these parameters? Clients generate their code from these specs and we probably don't want them generating with this parameter as an option.

Reply:
This comment goes for ALL the changed rest specs

Reply:
If we don't want the clients to include this option we should just remove it from these specs.

Reply:
Since most (if not all) clients are not BWC, it should probably be removed.

Reply:
I decided to keep those deprecated params as without them yml tests do not work and I do see a value in keeping the yml tests for those params.
Added `deprecated` flag so that those params stand out.